src/btree/bt_cursor.c

```diff
@@ -805,10 +844,18 @@ retry:	WT_ERR(__cursor_func_init(cbt, true));
 		 * column-store implicitly fills the gap with empty records.
 		 * Fail in that case, the record exists.
 		 */
-		if (!F_ISSET(cursor, WT_CURSTD_OVERWRITE) &&
-		    ((cbt->compare == 0 && __wt_cursor_valid(cbt, NULL)) ||
-		    (cbt->compare != 0 && __cursor_fix_implicit(btree, cbt))))
-			WT_ERR(WT_DUPLICATE_KEY);
+		if (!F_ISSET(cursor, WT_CURSTD_OVERWRITE)) {
+			if (cbt->compare ==0 ) {
+				visibility = __wt_cursor_valid(cbt, NULL);
```

Comment:
Same comments as the previous block. can combine assignment and comparison inside if block. Also needs looking into if rollback has to be returned.